pkg/helm/handlers/handlers.go

```diff
@@ -143,3 +151,79 @@ func (h *helmHandlers) HandleChartGet(user *auth.User, w http.ResponseWriter, r
 	res, _ := json.Marshal(resp)
 	w.Write(res)
 }
+
+func (h *helmHandlers) HandleUpgradeRelease(user *auth.User, w http.ResponseWriter, r *http.Request) {
+	var req HelmRequest
+
+	err := json.NewDecoder(r.Body).Decode(&req)
+	if err != nil {
+		serverutils.SendResponse(w, http.StatusBadRequest, serverutils.ApiError{Err: fmt.Sprintf("Failed to parse request: %v", err)})
+		return
+	}
+
+	conf := h.getActionConfigurations(h.ApiServerHost, req.Namespace, user.Token, &h.Transport)
+	resp, err := h.upgradeRelease(req.Namespace, req.Name, req.ChartUrl, req.Values, conf)
+	if err != nil {
+		serverutils.SendResponse(w, http.StatusBadGateway, serverutils.ApiError{Err: fmt.Sprintf("Failed to upgrade helm release: %v", err)})
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	res, _ := json.Marshal(resp)
+	w.Write(res)
+}
+
+func (h *helmHandlers) HandleUninstallRelease(user *auth.User, w http.ResponseWriter, r *http.Request) {
+	params := r.URL.Query()
+	ns := params.Get("ns")
+	rel := params.Get("name")
+
+	conf := h.getActionConfigurations(h.ApiServerHost, ns, user.Token, &h.Transport)
+	resp, err := h.uninstallRelease(rel, conf)
+	if err != nil {
+		serverutils.SendResponse(w, http.StatusBadGateway, serverutils.ApiError{Err: fmt.Sprintf("Failed to uninstall helm release: %v", err)})
+		return
+	}
+	w.Header().Set("Content-Type", "application/json")
+	res, _ := json.Marshal(resp)
+	w.Write(res)
+}
+
+func (h *helmHandlers) HandleRollbackRelease(user *auth.User, w http.ResponseWriter, r *http.Request) {
+	var req HelmRequest
+
+	err := json.NewDecoder(r.Body).Decode(&req)
+	if err != nil {
+		serverutils.SendResponse(w, http.StatusBadGateway, serverutils.ApiError{Err: fmt.Sprintf("Failed to parse request: %v", err)})
+		return
+	}
+
+	conf := h.getActionConfigurations(h.ApiServerHost, req.Namespace, user.Token, &h.Transport)
+	err = h.rollbackRelease(req.Name, req.Version, conf)
+	if err != nil {
+		serverutils.SendResponse(w, http.StatusBadGateway, serverutils.ApiError{Err: fmt.Sprintf("Failed to rollback helm releases: %v", err)})
+		return
+	}
+	resp := map[string]string{
```

Comment:
not sure what is the benefit of sending a human readable message in case of success. It would be better to send rollbacked release back?

Reply:
helm rollback api doesn't return rollbacked release, so to get it we would need to make extra helm get release call in rollback handler which might delay the response.

Reply:
Done